### ISSUE TITLE
Chocolatey hash checksum error

### ISSUE THREAD

**Issue by ThiloteE** [Betterbird/thunderbird-patches#185]
To install Betterbird on Windows 10, I used the [chocolatey](https://community.chocolatey.org/packages/betterbird) package manager.
Got this error during attempted upgrade of the Betterbird package.

I don't know, if Betterbird manages the BB chocolatey package, or some third party, but I thought it probably should be mentioned here.

```
C:\Windows\system32>choco upgrade betterbird
Chocolatey v2.2.0
Upgrading the following packages:
betterbird
By upgrading, you accept licenses for the packages.

You have betterbird v102.13.0 installed. Version 102.14.0 is available based on your source(s).
Progress: Downloading betterbird 102.14.0... 100%

betterbird v102.14.0 [Approved]
betterbird package files upgrade completed. Performing other installation steps.
Using locale 'en-US'...
File appears to be downloaded already. Verifying with package checksum to determine if it needs to be redownloaded.
Error - hashes do not match. Actual value was '3D24C286A7EFA961666F89F54CA9B2BF5DB0CB5DA623560851D827DA829D2E3F'.
Downloading betterbird
  from 'https://www.betterbird.eu/downloads/WindowsInstaller/betterbird-102.14.0-bb39.en-US.win64.installer.exe'
Progress: 100% - Completed download of c:\temp\choco\betterbird\102.14.0\betterbird-102.14.0-bb39.en-US.win64.installer.exe (52.21 MB).
Download of betterbird-102.14.0-bb39.en-US.win64.installer.exe (52.21 MB) completed.
Error - hashes do not match. Actual value was '3D24C286A7EFA961666F89F54CA9B2BF5DB0CB5DA623560851D827DA829D2E3F'.
ERROR: Checksum for 'c:\temp\choco\betterbird\102.14.0\betterbird-102.14.0-bb39.en-US.win64.installer.exe' did not meet '7946da74d9a34bb19eb501ff16f4d5459007180fdc693682187643684bb829cc' for checksum type 'sha256'. Consider passing the actual checksums through with --checksum --checksum64 once you validate the checksums are appropriate. A less secure option is to pass --ignore-checksums if necessary.
The upgrade of betterbird was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\betterbird\tools\chocolateyInstall.ps1'.
 See log for details.

Chocolatey upgraded 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - betterbird (exited -1) - Error while running 'C:\ProgramData\chocolatey\lib\betterbird\tools\chocolateyInstall.ps1'.
 See log for details.
```

**Comment by Betterbird** [comment]
Thanks for bringing the issue to our attention.

We are in contact with the FlatPak maintainer and the Arch Linux (AUR) maintainer and **no one else**. We are not really happy that third party sites are distributing our software without any prior contact and without any way to contact them when necessary, see below.

As for BB 102.14.0-bb39. We released it on 1st of August 2023 and shortly after we were alerted to an issue with NNTP/news. Sadly we made a mistake in applying code that worked in version 115 to version 102 where it didn't work (change of API name). So we had to completely replace 102.14.0.

The checksum changed from 7946da74d9a34bb19eb501ff16f4d5459007180fdc693682187643684bb829cc to 3d24c286a7efa961666f89f54ca9b2bf5db0cb5da623560851d827da829d2e3f for the Windows installer.

Maybe this line "File appears to be downloaded already." gives a hint. Does the situation improve if you delete your local download? Can you run `sha256sum` on it?

We're closing this issue since we can't action it.

**Comment by ThiloteE** [comment]
The file was downloaded already, because it was downloaded the first time I attempted the upgrade process, which failed and the error message I sent you was from my second attempt at upgrading.

I checked the temp folder and executing `sha256sum` on the file `betterbird-102.14.0-bb39.en-US.win64.installer` indeed outputs the hash `3d24c286a7efa961666f89f54ca9b2bf5db0cb5da623560851d827da829d2e3f`, which should be correct.

I then executed 

`choco upgrade betterbird --checksum 3d24c286a7efa961666f89f54ca9b2bf5db0cb5da623560851d827da829d2e3f` and the upgrade was then successful.
